int32 id
int32 aid
int32 type # Don't use wide character
int32 linkid
